src/server/services/storage_portal/storage_portal_memory.ts

```diff
@@ -118,7 +126,14 @@ export default class MemoryStoragePortal implements StoragePortal {
     passthrough.on("data", (chunk) => {
       this.buckets[bucket] = this.buckets[bucket] || {};
       this.buckets[bucket][objectKey] = this.buckets[bucket][objectKey] || {};
-      this.buckets[bucket][objectKey].contents += chunk;
+      if (!this.buckets[bucket][objectKey].contents) {
```

Comment:
There's many times when we use the phrase 'this.buckets[bucket][objectKey]', it may be nicer to assign it to a temp variable for ease of reading & for tweaking in the future.
I'm unsure if there's enough uses of the phrase to make it worth aliasing though.

Reply:
Great idea, I added a helper to do the initialization and retrieval of the object info.